hail/src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -33,168 +33,4 @@ class HtsjdkRecordReader(
            |  expected ${TFloat32()._toPretty} or ${TFloat64()._toPretty}, got ${entryFloatTypeName}"""
     )
   }
-
-  def readVariantInfo(
-    vc: VariantContext,
-    rvb: RegionValueBuilder,
-    hasRSID: Boolean,
-    hasQual: Boolean,
-    hasFilters: Boolean,
-    infoType: TStruct,
-    infoFlagFieldNames: Set[String]) {
-    // locus, alleles added via VCFLine
-
-    // rsid
-    if (hasRSID) {
-      val vcID = vc.getID
-      if (vcID == ".")
-        rvb.setMissing()
-      else
-        rvb.addString(vcID)
-    }
-
-    if (hasQual) {
-      rvb.addDouble(vc.getPhredScaledQual)
-    }
-
-    // filters
-    if (hasFilters) {
-      if (!vc.filtersWereApplied)
-        rvb.setMissing()
-      else {
-        if (vc.isNotFiltered) {
-          rvb.startArray(0)
-          rvb.endArray()
-        } else {
-          rvb.startArray(vc.getFilters.size())
-          val fi = vc.getFilters.iterator()
-          while (fi.hasNext)
-            rvb.addString(fi.next())
-          rvb.endArray()
-        }
-      }
-    }
-
-    if (infoType != null) {
-      rvb.startStruct()
-      infoType.fields.foreach { f =>
-        val a = vc.getAttribute(f.name)
-        addAttribute(rvb, a, f.typ, -1, isFlag = infoFlagFieldNames.contains(f.name))
-      }
-      rvb.endStruct()
-    }
-  }
-}
-
-object HtsjdkRecordReader {
-  def addAttribute(rvb: RegionValueBuilder, attr: Any, t: Type, nAlleles: Int, isFlag: Boolean = false) {
-    ((attr, t): @unchecked) match {
-      case (null, _) =>
-        if (isFlag)
-          rvb.addBoolean(false)
-        else
-          rvb.setMissing()
-      case (".", _) =>
-        rvb.setMissing()
-      case (s: String, TArray(_: TInt32, _)) =>
-        val xs = s.split(",")
-        rvb.startArray(xs.length)
-        xs.foreach { x =>
-          if (x == ".")
-            rvb.setMissing()
-          else
-            rvb.addInt(x.toInt)
-        }
-        rvb.endArray()
-
-      case (s: String, TArray(_: TFloat64, _)) =>
-        val xs = s.split(",")
-        rvb.startArray(xs.length)
-        xs.foreach { x =>
-          if (x == ".")
-            rvb.setMissing()
-          else
-            rvb.addDouble(x.toDouble)
-        }
-        rvb.endArray()
-      case (s: String, TArray(_: TString, _)) =>
-        val xs = s.split(",")
-        rvb.startArray(xs.length)
-        xs.foreach { x =>
-          if (x == ".")
-            rvb.setMissing()
-          else
-            rvb.addString(x)
-        }
-        rvb.endArray()
-      case (s: String, _: TBoolean) =>
-        rvb.addBoolean(s.toBoolean)
-      case (b: Boolean, _: TBoolean) =>
-        rvb.addBoolean(b)
-      case (s: String, _: TString) =>
-        rvb.addString(s)
-      case (s: String, _: TInt32) =>
-        rvb.addInt(s.toInt)
-      case (s: String, _: TFloat64) =>
-        val d = s match {
-          case "nan" => Double.NaN
-          case "-nan" => Double.NaN
```

Comment:
I don't think your float parsing handles these correctly.

Reply:
You are correct. Right now it handles it exactly as java would, to be properly VCF 4.3 compliant, we would need to handle, `NaN`, `Inf`, and `-Inf`. Which we don't even handle properly right now, just their lower case versions.

Reply:
Can you make an issue then?

Reply:
Futhermore, VCF 4.2 makes no claims about the string representation of floats at all. VCF is garbage.